### PR TITLE
fix(ext/node): add util.parseArgs

### LINKED ISSUE #20452: [feature-request] `node:util`'s parseArgs
I would like to have node:util parseArgs for Deno as well.

https://nodejs.org/api/util.html#utilparseargsconfig

```ts
import { parseArgs } from "node:util";
const args = parseArgs({
  args: Deno.args,
  options: {
    flag: {
      type: "boolean",
      alias: "f",
    }
  },
  allowPositionals: true,
});

// @ts-expect-error
const f: string = args.values.flag;
```

## Why

- Standard since node 18.3
- It has a good interface for TypeScript inference

I often use deno to make my own little CLI tools, and I tried to use the parseArgs I'm familiar with in deno, but it doesn't exist in deno's node:util.

Of course, I understand that there is a flags module in deno.land/std and the interface is just a matter of preference. But if deno claims to be compatible with node, it would be nice to have it.

### PR DESCRIPTION
**Pull request by kt3k**
This PR adds [`parseArgs`](https://nodejs.org/api/util.html#utilparseargsconfig) of `node:util`.

```js
import { parseArgs } from "node:util";
const args = ['-f', '--bar', 'b'];
const options = {
  foo: {
    type: 'boolean',
    short: 'f',
  },
  bar: {
    type: 'string',
  },
};
const {
  values,
  positionals,
} = parseArgs({ args, options });
console.log(values, positionals);
```

```shellsession
$ ./target/debug/deno run a.mjs              
[Object: null prototype] { foo: true, bar: "b" } []
```

closes #20452

### CODE REVIEW

File: ext/node/polyfills/internal/util/parse_args/parse_args.js [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+const primordials = globalThis.__bootstrap.primordials;
+const {
+  ArrayPrototypeForEach,
+  ArrayPrototypeIncludes,
+  ArrayPrototypePushApply,
+  ArrayPrototypeShift,
+  ArrayPrototypeSlice,
+  ArrayPrototypePush,
+  ArrayPrototypeUnshiftApply,
+  ObjectPrototypeHasOwnProperty: ObjectHasOwn,
```
Comment (petamoriken) [inline ext/node/polyfills/internal/util/parse_args/parse_args.js:13]: This alias is not necessary

```suggestion
  ObjectHasOwn,
```

File: ext/node/polyfills/internal/util/parse_args/utils.js [hunk 2]
```diff
@@ -0,0 +1,187 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+// Copyright Joyent and Node contributors. All rights reserved. MIT license.
+
+const primordials = globalThis.__bootstrap.primordials;
+const {
+  ArrayPrototypeFind,
+  ObjectEntries,
+  ObjectPrototypeHasOwnProperty: ObjectHasOwn,
```
Comment (petamoriken) [inline ext/node/polyfills/internal/util/parse_args/utils.js:8]: ditto

File: ext/node/polyfills/internal/util/parse_args/parse_args.js (hunk 1 above)
Comment (kt3k) [reply ext/node/polyfills/internal/util/parse_args/parse_args.js]: Makes sense. Thanks

File: cli/tests/node_compat/test.ts [hunk 3]
```diff
@@ -83,6 +83,7 @@ async function runTest(t: Deno.TestContext, path: string): Promise<void> {
         "--quiet",
         "--unstable",
         //"--unsafely-ignore-certificate-errors",
+        "--unstable-bare-node-builtins",
```
Comment (bartlomieju) [inline cli/tests/node_compat/test.ts:86]: Why is this flag required?

Comment (kt3k) [reply cli/tests/node_compat/test.ts:86]: The added case (`test-parse-args.mjs`) is the first one in our compat test suite that depends on  `cli/tests/node_compat/test/common/index.mjs`, and that file has bare import of `module` module https://github.com/denoland/deno/blob/45fe3ccccfc2db9a969d2e6d258011ae96ee7d1e/cli/tests/node_compat/test/common/index.mjs#L8

**Review by bartlomieju — APPROVED** [review]
LGTM, nice work. Please update compat list in `deno-docs` repo after landing.

**Comment by kt3k** [comment]
> Please update compat list in deno-docs repo after landing.

I opened https://github.com/denoland/deno-docs/pull/206 and https://github.com/denoland/deno/issues/21378. There seem still 7 APIs missing in node:util.

**Pull request merged**